packages/core/src/routes/hook.ts

```diff
@@ -21,23 +24,59 @@ export default function hookRoutes<T extends AuthedRouter>(
     }
   );
 
+  router.get(
+    '/hooks/:id',
+    koaGuard({
+      params: z.object({ id: z.string() }),
+      response: Hooks.guard,
+      status: [200, 404],
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+
+      ctx.body = await findHookById(id);
+
+      return next();
+    }
+  );
+
   router.post(
     '/hooks',
-    koaGuard({ body: Hooks.createGuard.omit({ id: true }), response: Hooks.guard, status: 200 }),
+    koaGuard({
+      body: Hooks.createGuard
+        .omit({ id: true, name: true, signingKey: true, enabled: true, events: true })
+        .extend({
+          // Note: ensure the user will not create a hook with an empty name.
+          name: string().min(1).optional(),
+          events: hookEventsGuard.nonempty().optional(),
+        }),
+      response: Hooks.guard,
+      status: [201, 422],
+    }),
     async (ctx, next) => {
+      const { event, events, ...rest } = ctx.guard.body;
+      assertThat(events ?? event, new RequestError({ code: 'hook.missing_events', status: 422 }));
+
       ctx.body = await insertHook({
+        ...rest,
         id: generateStandardId(),
-        ...ctx.guard.body,
+        signingKey: generateStandardId(),
+        enabled: true,
+        ...conditional(event && { event }),
+        ...conditional(events && { events: deduplicate(events) }),
```

Comment:
per discussion should we just ignore `event` if `events` presents?

Reply:
I think it's ok to keep the `event` value since if a user is always using the old API, he may want to see the `event` being updated.